javascript/node/selenium-webdriver/lib/webdriver.js

```diff
@@ -786,11 +786,8 @@ class WebDriver {
         let start = Date.now();
         let timer = setTimeout(function() {
           timer = null;
-          reject(
-              new error.TimeoutError(
-                  (message ? `${message}\n` : '')
-                      + 'Timed out waiting for promise to resolve after '
-                      + (Date.now() - start) + 'ms'));
+          let timeoutMessage = message ? `${typeof message === 'function' ? message() : message}\n` : '';
```

Comment:
If the message function throws, the promise will never be resolved. So you probably need something like:

```
try {
  let msg = message();  // Shortened for brevity
  reject(new error.TimeoutError(msg));
} catch (ex) {
  reject(new error.TimeoutError(`${ex}\nTimed out waiting...`));
}
```

Reply:
@jleyba good idea. updated the PR with additional tests for these scenarios.